src/50_Samples_%26_Templates/Product.html

```diff
@@ -335,69 +335,71 @@ <h2 class="m1">Apple</h2>
+        <amp-img on="tap:AMP.setState({product: {selectedSlide: 0}})"
                   src="/img/green_apple_1_60x40.jpg"
                   width="60" height="40"
                   class="selected"
-                  [class]="selectedSlide == 0 ? 'selected' : '' ">
+                  [class]="product.selectedSlide == 0 ? 'selected' : '' ">
         </amp-img>
       </li>
       <li tabindex="1" role="button">
-        <amp-img on="tap:AMP.setState({selectedSlide: 1})"
+        <amp-img on="tap:AMP.setState({product: {selectedSlide: 1}})"
                   src="/img/green_apple_2_60x40.jpg"
                   width="60" height="40"
-                  [class]="selectedSlide == 1 ? 'selected' : '' ">
+                  [class]="product.selectedSlide == 1 ? 'selected' : '' ">
         </amp-img>
       </li>
     </ul>
 
-    <ul class="hide" [class]="selectedColor == 'golden' ? 'show' : 'hide'">
+    <ul class="hide" [class]="product.selectedColor == 'golden' ? 'show' : 'hide'">
       <li tabindex="0" role="button">
-        <amp-img on="tap:AMP.setState({selectedSlide: 0})"
+        <amp-img on="tap:AMP.setState({product: {selectedSlide: 0}})"
                 src="/img/product1_alt1_60x40.jpg"
                  width="60" height="40"
-                 [class]="selectedSlide == 0 ? 'selected' : '' ">
+                 [class]="product.selectedSlide == 0 ? 'selected' : '' ">
         </amp-img>
       </li>
     </ul>
 
-    <ul class="hide" [class]="selectedColor == 'red' ? 'show' : 'hide'">
+    <ul class="hide" [class]="product.selectedColor == 'red' ? 'show' : 'hide'">
       <li tabindex="0" role="button">
-        <amp-img on="tap:AMP.setState({selectedSlide: 0})"
+        <amp-img on="tap:AMP.setState({product: {selectedSlide : 0}})"
                  src="/img/red_apple_1_60x40.jpg"
                  width="60" height="40"
-                 [class]="selectedSlide == 0 ? 'selected' : '' ">
+                 [class]="product.selectedSlide == 0 ? 'selected' : '' ">
         </amp-img>
       </li>
       <li tabindex="1" role="button">
-        <amp-img on="tap:AMP.setState({selectedSlide: 1})"
+        <amp-img on="tap:AMP.setState({product: {selectedSlide : 1}})"
                  src="/img/red_apple_2_60x46.jpg"
                  width="60" height="40"
-                 [class]="selectedSlide == 1 ? 'selected' : '' ">
+                 [class]="product.selectedSlide == 1 ? 'selected' : '' ">
         </amp-img>
       </li>
     </ul>
   </div>
    <!-- ## Image Gallery -->
   <!-- The `amp-carousel` component works very well for product image galleries. Learn how the `amp-carousel` component works [here](/advanced/image_galleries_with_amp-carousel/).
 
-  Notice how we are binding to the `slide` and `class` by using to the values of the variables `selectedSlide` and `selectedColor`. -->
+  Notice how we are binding to the `slide` and `class` by using to the values of the variables `selectedSlide` and `selectedColor`.
```

Comment:
I think this paragraph needs to be changed to discuss the new approach of using an `amp-state` to initialize these variables.